src/__experimental__/components/date/date.spec.js

```diff
@@ -569,6 +570,16 @@ describe('Date', () => {
   });
 });
 
+describe('when the calendar icon is clicked and the picker is open', () => {
```

Comment:
is there alternative to that? Maybe worth adding test the behaviour of two clicks. 1st click closes the picker, 2nd click re-opens the picker?